clippy_lints/src/entry.rs

```diff
@@ -142,13 +145,21 @@ impl<'a, 'tcx, 'b> Visitor<'tcx> for InsertVisitor<'a, 'tcx, 'b> {
             if SpanlessEq::new(self.cx).eq_expr(self.key, &params[1]);
             if snippet_opt(self.cx, self.map.span) == snippet_opt(self.cx, params[0].span);
             then {
+                // if blocks without an `else` never need a semicolon after them,
+                // but we're converting them to call expressions which do
+                let maybe_semi = if self.has_else {
+                    ""
+                } else {
+                    ";"
+                };
                 span_lint_and_then(self.cx, MAP_ENTRY, self.span,
                                    &format!("usage of `contains_key` followed by `insert` on a `{}`", self.ty), |db| {
                     if self.sole_expr {
-                        let help = format!("{}.entry({}).or_insert({})",
+                        let help = format!("{}.entry({}).or_insert({}){}",
```

Comment:
Dogfood says that these two if branches are now identical. Is that what we want? What was the reason for not including `or_insert({})` in the `else` branch of the original code?

Reply:
Ah, I should be splitting this file, I made a mistake

Reply:
Removed the commit, will deal with later.

Reply:
We need multiple suggestions for the other one, it's going to be tricky